File: src/javascript/user/message/update-tracking-record.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const messageID = process.env.FAXCORE_MESSAGE_ID;
const trackID = process.env.FAXCORE_TRACK_ID;
const trackValue = process.env.FAXCORE_TRACK_VALUE;

if (!messageID || !trackID || !trackValue) {
  throw new Error('FAXCORE_MESSAGE_ID, FAXCORE_TRACK_ID, and FAXCORE_TRACK_VALUE are required.');
}

const client = new FaxCoreClient();
const result = await client.put('/api/message/tracking', {
  messageID,
  trackID: Number(trackID),
  trackValue
});

printResult('Update tracking record', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/UpdateTrackingRecord.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class UpdateTrackingRecord
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (string.IsNullOrWhiteSpace(config.MessageId) || !config.TrackID.HasValue || string.IsNullOrWhiteSpace(config.TrackValue))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID, FAXCORE_TRACK_ID, and FAXCORE_TRACK_VALUE before running user.message.tracking.update.");
            }

            return client.PutAsync("/api/message/tracking", new
            {
                messageID = config.MessageId,
                trackID = config.TrackID.Value,
                trackValue = config.TrackValue
            });
        }
    }
}

